cwltool/main.py

```diff
@@ -695,6 +697,18 @@ def main(argsl=None,  # type: List[str]
                 argsl = sys.argv[1:]
             args = arg_parser().parse_args(argsl)
 
+        # If On windows platform, A default Docker Container is Used if not explicitely provided by user
+        if onWindows() and not args.default_container:
+            args.default_container = "ubuntu"
+
+        # Making default Container as lowest priority, that should be used if nothing overrides it
+        if args.default_container:
```

Comment:
One more round of refactoring will be needed for this. I would create a `get_docker_requirement` function that should include result from default_container also if result from hints and requirement is false.